en/02_Developer_Guides/14_Files/05_File_Manipulation.md

```diff
@@ -0,0 +1,419 @@
+| `File::getURL`           | Gets the URL for this resource. May or may not be absolute |
+| `File::getAbsoluteURL`   | Gets the absolute URL to this resource                     |
+| `File::getMimeType`      | Get the mime type of this file                             |
+| `File::getMetaData`      | Gets other metadata from the file as an array              |
+| `File::getFileType`      | Return the type of file for the given extension            |
+
+### Additional file types
+
+Silverstripe CMS has a pre-defined list of common file types. `File::getFileType` will return "unknown" for files outside that list.
+
+You can add your own file extensions and its description with the following configuration.
+
+```yml
+SilverStripe\Assets\File:
+  file_types:
+    ai: 'Adobe Illustrator'
+    psd: 'Adobe Photoshop File'
+```
+
+## Renaming and moving files
+
+In order to move or rename a file you can simply update the `Name` property, or assign the `ParentID` to a new
+folder. Please note that these modifications are made simply on the draft stage, and will not be copied
+to live until a publish is made via the CMS (either on this object, or cascading from a parent).
+
+When files are renamed using the ORM, all file variants are automatically renamed at the same time.
+
+```php
+use SilverStripe\Assets\File;
+
+$file = File::get()->filter('Name', 'oldname.jpg')->first();
+if ($file) {
+  // The below will move 'oldname.jpg' and 'oldname__variant.jpg'
+  // to 'newname.jpg' and 'newname__variant.jpg' respectively
+    $file->Name = 'newname.jpg';
+    $file->write();
+}
+```
+
+Note that you can cause the file to be moved immediately by [setting the Versioned reading mode](api:SilverStripe\Versioned\Versioned::set_reading_mode()) to draft temporarily.
+
+```php
+use SilverStripe\Assets\File;
+use SilverStripe\Versioned\Versioned;
+
+$file = File::get()->filter('Name', 'oldname.jpg')->first();
+if ($file) {
+  // The below will immediately move 'oldname.jpg' and 'oldname__variant.jpg'
+  // to 'newname.jpg' and 'newname__variant.jpg' respectively
+    $file->Name = 'newname.jpg';
+    Versioned::withVersionedMode(function () use ($file) {
+        Versioned::set_reading_mode('Stage.' . Versioned::DRAFT);
+        $file->write();
+        $file->publishSingle();
+    });
+}
+```
+
+## Convert a file to a different format
+
+You can use the [`manipulateExtension()`](api:SilverStripe\Assets\ImageManipulation::manipulateExtension()) method on any `File` or `DBFile` object to create a variant with a different file extension than the original.
+
+This can be very useful if you want to convert a file to a different format for the user to download or view, while leaving the original file intact. Some examples of when you might want this are:
+
+- Generating thumbnails for videos, documents, etc
+- Converting images to to `.webp` for faster page load times
+- Converting documents to `.pdf` so downloaded documents are more portable
```

Comment:
```suggestion
- Generating thumbnails for videos, documents, etc.
- Converting images to `.webp` for faster page load times
- Converting documents to `.pdf` so downloaded documents are more portable
```

Reply:
Removed the extra "to" - won't add the fullstop because we're not adding it to the other list items.